src/SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -128,17 +128,18 @@ public override bool Execute()
             if (guid != null && Guid.TryParse(guid, out Guid projectId))
             {
                 pi.ProjectGuid = projectId;
-                pi.AnalysisResults = TryCreateAnalysisResults(AnalysisResults);
-                pi.AnalysisSettings = TryCreateAnalysisSettings(AnalysisSettings);
-
-                var outputFileName = Path.Combine(OutputFolder, FileConstants.ProjectInfoFileName);
-                pi.Save(outputFileName);
             }
             else
             {
                 Log.LogWarning(Resources.WPIF_MissingOrInvalidProjectGuid, FullProjectPath);
             }
 
+            pi.AnalysisResults = TryCreateAnalysisResults(AnalysisResults);
```

Comment:
We now create the `ProjectInfo.xml` file for projects without any GUID and we do the filtering later on. We still display the build warning.